lib/step-value-collection.js

```diff
@@ -0,0 +1,43 @@
+'use strict';
+
+const inherits      = require('util').inherits;
+const EventEmitter  = require('events').EventEmitter;
+
+class StepValueCollection {
+  constructor(collection) {
+    this.collection = collection;
+  }
+
+  valid() {
+    return this.collection.every((item) => { return item.valid(); });
+  }
+
+  ambiguousSteps() {
+    return this.collection.filter((item) => { return item.ambiguous(); });
+  }
+
+  notFoundSteps() {
+    return this.collection.filter((item) => { return item.notFound(); });
+  }
+
+  map(iterator) {
+    return this.collection.map(iterator);
+  }
+
+  listen(listener) {
+    this.collection.forEach((item) => {
```

Comment:
I thought the es6 arrow functions handled this. I have been running more tests, and not having the usual deref issues that I experience with JS